docs/developer-docs/build/backend/reproducible-builds.md

```diff
@@ -124,70 +127,48 @@ For the build process to be deterministic:
+Now, from the root directory of your canister project, you can test the reproducibility of your `dfx` builds as follows:
 
     $ docker build -t mycanister .
     ...
     $ docker run --rm --privileged -it mycanister
-    root@6fe19d89f8f5:/canister# reprotest -vv "dfx build --network ic" '.dfx/ic/canisters/*/*.wasm'
+    root@6fe19d89f8f5:/canister# reprotest -vv --variations '+all,-time' "dfx build --network ic" '.dfx/ic/canisters/*/*.wasm'
 
-The first command builds the Docker canister using the above `Dockerfile`. The second one opens an interactive shell (hence the `-it` flags) in the canister. We run this in privileged mode (the `--privileged` flag), as `reprotest` uses kernel modules for some build environment variations. You can also run it in non-privileged mode by excluding some of the variations; see the [reprotest manual](https://manpages.debian.org/stretch/reprotest/reprotest.1.en.html). The `--rm` flag will destroy the canister after you close its shell. Finally, once inside of the canister, we launch `reprotest` in verbose mode (the `-vv` flags). You need to give it the build command you want to run as the first argument. Here, we assume that it’s `dfx build --network ic` - adjust it if you’re using a different build process. It will then run the build in two different environments. Finally, you need to tell `reprotest` which paths to compare at the end of the two builds. Here, we compare the Wasm code for all canisters, which is found in the `.dfx/ic` directory.
+The first command builds the Docker container using the above `Dockerfile`. The second one opens an interactive shell (hence the `-it` flags) in the canister. We run this in privileged mode (the `--privileged` flag), as `reprotest` uses kernel modules for some build environment variations. You can also run it in non-privileged mode by excluding some of the variations; see the [reprotest manual](https://manpages.debian.org/stretch/reprotest/reprotest.1.en.html). We exclude the `time` variation for Rust builds (we observed that Rust builds crashed upon the `time` variation). The `reprotest` docs say that "the `time` variation uses faketime which sometimes causes weird and hard-to-diagnose problems" so it might not be straightforward to enable the `time` variations for Rust builds. For Motoko builds, you could omit `--variations '+all,-time'` and thereby trying out all supported variations. The `--rm` flag will destroy the canister after you close its shell. Finally, once inside of the canister, we launch `reprotest` in verbose mode (the `-vv` flags). You need to give it the build command you want to run as the first argument. Here, we assume that it’s `dfx build --network ic` - adjust it if you’re using a different build process. It will then run the build in two different environments. Finally, you need to tell `reprotest` which paths to compare at the end of the two builds. Here, we compare the Wasm code for all canisters, which is found in the `.dfx/ic` directory.
```

Comment:
This is a bit worrying - I did not have problems with Rust builds when varying time. Could you share how you encountered the problem? Generally, invariance under time looks like a pretty basic desirable property to check, as it's easy to time to leak in.

Nit: "above `Dockerfile`" doesn't apply now as it's not listed above any more.

Reply:
Filed an issue with faketime: https://github.com/wolfcw/libfaketime/issues/402

Reply:
Hi @oggy-dfin! Are you happy with this enough to resolve and merge?

Reply:
Hi @aisconnolly - I'd like to get https://github.com/mraszyk/portal/pull/2 merged first, after that it's good from my side

Reply:
Merged!